client/src/main/java/io/pravega/client/ClientConfig.java

```diff
@@ -101,6 +103,21 @@
     @Getter(AccessLevel.NONE) // Omit Lombok accessor as we are creating a custom one
     private final boolean enableTlsToSegmentStore;
 
+
+    /**
+     * A optional listener which can be used to get performance metrics from the client. The user
+     * can implement this interface to obtain performance metrics of the client.
+     */
+    private final MetricListener metricListener;
+
+    /**
+     * A method which returns the optional metric listener {@link MetricListener} .
+     * @return MetricListener.
+     */
+    public Optional<MetricListener> getMetricListener() {
```

Comment:
What's the difference between a plain getter that can return `null` and returning an `Optional` which you can check with null? IMO, the plain getter is simpler to implement and to use, especially for nullable values. Plus you don't have to carry an `Optional` everywhere you pass this (constructors, etc.)